### PR TITLE
Improve startup time of applications

### PR DESCRIPTION
**Pull request by GernotMaier**
Applications are very slow in starting up, see issue #681 .

This is mostly related to imports and this pull request is start to optimize this:

- moved hd5-related functions out of utils/general.py into a separate module (io_hdf5)
- for clarity, moved io_handler and io_hdf5 into a new directory simtools.io_operations (most changes of this PR are related to the change of the import statement; note that I have in an earlier version this was simtools.io, but I realised that io is in conflict with the python io package)

This improves the speed of calling e.g., `python simtools/applications/validate_schema_files.py --help` by a factor of 4!

### CODE REVIEW

File: simtools/io_operations/io_hdf5.py
Comment (VictorBarbosaMartins) [inline simtools/io_operations/io_hdf5.py:19]: It is good to have all the hdf5 functions together, but this function is not actually an io as it does not deal with input and output. Instead, if fills the table. Maybe we should call the module `hdf5_handler` instead?

**Review by VictorBarbosaMartins — APPROVED** [review]
Thanks for these updates, it is improving also in the direction of #665.
I left only one comment/suggestion.

File: simtools/io_operations/io_hdf5.py
Comment (GernotMaier) [reply simtools/io_operations/io_hdf5.py]: Thanks - this is a good suggestion. I've renamed `io_hdf5` to `hd5_handler`.

Waiting for the tests and will merge when successful.

**Pull request merged**
